### PR TITLE
Support Angular inline styles as single template literal

### LINKED ISSUE #15934: Angular inline styles are not formatted if specified as single string (new in v17)
**Prettier 3.2.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAmAGEItoEYBffAMx1PwHIABAQygHMBXAG1ZwD0kHHEYBuADpQpzEmSgUAFMCn58AZzg84YGLiRNWmTAFp6EGIwA0q-PCx94BgAYAJLTwjObUNepgAntrqLrZqSAAWEP4qvmrxkJ44SABGPFxwYpS23lKUAJRScAAe2Hj4YHzq6vgAgsZy2AqwhNlQIFYgEJgwqNDqyKD89ADuAAr8CAMorDwjrAEDnSk4rGAA1nAwAMqs6HAAMqgKyDSzmsurG1vbmGvHHMgwOBmdmuioTy9wnSWYcDgMBRZgAVAFQfioODTM48C4gdQPbQARS4Fjgp3OPxAACt1MVtki4Kj0Zi4diAI5o+BjeiYaYgVjqEwKOAAE3ZHRAz1YqB4Dzk6FYyEZPB4XMRnG0tRgz1QKS4NIBRxOSFh8IiMHQPAA6hFUPB1HcwHBtlMDagAG4GgIisDVLmWjIASSgHNg2zAgJ6tTd20C2jJ8Mw9E0OtWmBFIehAMtGM6x00eFprA4QqD2LuOCTIpSrBSWi5IeOMB1qDZMAiyAAHAAGToiKmoEQptPCtVYzowfNlitVpAAJk6XE0IPzMM7IDg6ALbI5bIO7G4qbgADFcELZQ8RaxFRAQJRKEA)

```sh
--parser babel
```

**Input:**

```jsx
import {Component} from '@angular/core';

@Component({
  selector: 'app-root',
  template: `Hello`,
  styles: `
    :host{
      color:blue;}
  `,
})
export class AppComponent {}

```

**Output:**

```jsx
import { Component } from "@angular/core";

@Component({
  selector: "app-root",
  template: `Hello`,
  styles: `
    :host{
      color:blue;}
  `,
})
export class AppComponent {}

```

**Expected output:**

```jsx

```

**Why?**

Since Angular v17 it is possible to specify an inline style with a single string rather than an array of strings. If specified as array prettier formats the css. If specified as string prettier does nothing with the css.

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Closes #15934

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/embed/utils.js [hunk 1]
```diff
@@ -30,15 +30,24 @@ const angularComponentObjectExpressionPredicates = [
...
-      isObjectProperty(node) &&
-      node.key.type === "Identifier" &&
-      node.key.name === "styles" &&
-      name === "value",
-    ...angularComponentObjectExpressionPredicates,
+  const isTemplateLiteral = (node) => node.type === "TemplateLiteral";
+  const isStyleProperty = (node, name) =>
+    isObjectProperty(node) &&
+    node.key.type === "Identifier" &&
+    node.key.name === "styles" &&
```
Comment (fisker) [inline src/language-js/embed/utils.js:37]: `!node.computed`

File: src/language-js/embed/utils.js [hunk 2]
```diff
@@ -30,15 +30,24 @@ const angularComponentObjectExpressionPredicates = [
...
-    (node) => node.type === "TemplateLiteral",
-    (node, name) => isArrayOrTupleExpression(node) && name === "elements",
-    (node, name) =>
-      isObjectProperty(node) &&
-      node.key.type === "Identifier" &&
-      node.key.name === "styles" &&
-      name === "value",
-    ...angularComponentObjectExpressionPredicates,
+  const isTemplateLiteral = (node) => node.type === "TemplateLiteral";
+  const isStyleProperty = (node, name) =>
```
Comment (fisker) [inline src/language-js/embed/utils.js:34]: How about `isObjectPropertyNamedStyles`?

Comment (fisker) [reply src/language-js/embed/utils.js:34]: Second parameter should be named `key`

File: src/language-js/embed/utils.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-js/embed/utils.js:38]: I fixed this and added tests. Also I've create the new similar issue for `template` as a good first ( https://github.com/prettier/prettier/issues/15969 )

**Review by fisker — APPROVED** [review]

**Pull request merged**